tool/gravity/cli/config.go

```diff
@@ -346,56 +347,78 @@ func (i *InstallConfig) ToInstallerConfig(env *localenv.LocalEnvironment) (*inst
 		Insecure:           i.Insecure,
 		Manual:             i.Manual,
 		ServiceUser:        i.ServiceUser,
-		GCENodeTags:        i.NodeTags,
 		NewProcess:         i.NewProcess,
 		LocalClusterClient: env.SiteOperator,
+		RuntimeResources:   kubernetesResources,
+		ClusterResources:   gravityResources,
 	}, nil
 }
 
-// ValidateResources validates the resources specified in ResourcePath
-// using the given validator
-func (i *InstallConfig) ValidateResources(validator resources.Validator) error {
+// splitResources validates the resources specified in ResourcePath
+// using the given validator and splits them into Kubernetes and Gravity-specific
+func (i *InstallConfig) splitResources(validator resources.Validator) ([]runtime.Object, []storage.UnknownResource, error) {
 	if i.ResourcesPath == "" {
-		return trace.NotFound("no resources provided")
+		return nil, nil, trace.NotFound("no resources provided")
 	}
 	rc, err := utils.ReaderForPath(i.ResourcesPath)
 	if err != nil {
-		return trace.Wrap(err, "failed to read resources")
+		return nil, nil, trace.Wrap(err, "failed to read resources")
 	}
 	defer rc.Close()
 	// TODO(dmitri): validate kubernetes resources as well
-	_, gravityResources, err := resources.Split(rc)
+	kubernetesResources, gravityResources, err := resources.Split(rc)
 	if err != nil {
-		return trace.BadParameter("failed to validate %q: %v", i.ResourcesPath, err)
+		return nil, nil, trace.BadParameter("failed to validate %q: %v", i.ResourcesPath, err)
 	}
 	for _, res := range gravityResources {
 		log.WithField("resource", res.ResourceHeader).Info("Validating.")
 		if err := validator.Validate(res); err != nil {
-			return trace.Wrap(err, "resource %q is invalid", res.Kind)
+			return nil, nil, trace.Wrap(err, "resource %q is invalid", res.Kind)
 		}
 	}
-	return nil
+	return kubernetesResources, gravityResources, nil
 }
 
-func (i *InstallConfig) updateFromClusterConfig(resourceBytes []byte) error {
-	if len(resourceBytes) == 0 {
-		return nil
-	}
-	err := resources.ForEach(bytes.NewReader(resourceBytes), func(res storage.UnknownResource) error {
-		if res.Kind != storage.KindClusterConfiguration {
-			return nil
+func (i *InstallConfig) updateClusterConfig(resources []storage.UnknownResource) (updated []storage.UnknownResource, err error) {
+	if len(resources) == 0 {
+		return nil, nil
+	}
+	var clusterConfig *storage.UnknownResource
+	updated = resources[:0]
```

Comment:
Weird way to declare an empty slice, wouldn't `updated = make([]storage.UnknownResource, 0, len(resources))` do?

Reply:
It's a way of reusing a throw-away space